### ISSUE TITLE
types.extends throws

### ISSUE THREAD

**Issue by RainerAtSpirit** [mobxjs/mobx-state-tree#92]
Given a factory like
```javascript
const ComplexComputedFactory = types.model({
        width: 100,
        height: 200,
        stringifiedJSON: '{"a":{"b":{"c":1,"d":2}}}',
        get jsonObj(){
            return JSON.parse(this.stringifiedJSON)
        }
    })

...
test("it should not throw with complex computeds", (t) => {
    const {ComplexComputedFactory} = createTestFactories()
  
    t.deepEqual(getSnapshot(ComplexComputedFactory.create()), {width: 100, height: 200, stringifiedJSON: "{\"a\":{\"b\":{\"c\":1,\"d\":2}}}"})
})

test("it should compose factories with complex computeds return types", (t) => {
    const {BoxFactory, ComplexComputedFactory} = createTestFactories()
    const ComposedFactory = types.extend(BoxFactory, ComplexComputedFactory)
    
    t.deepEqual(getSnapshot(ComposedFactory.create()), {width: 100, height: 200, stringifiedJSON: '{"a":{"b":{"c":1,"d":2}}}'})
})
```

Using the Factory standalone works as expected, but when trying to compose the system throws
`[Error: [mobx-state-tree] In property 'jsonObj': base model's should not contain complex values: '[object Object]']`

https://github.com/mobxjs/mobx-state-tree/compare/master...RainerAtSpirit:composeFactories#diff-3a20877e2af1ada02a42d389756a845aR252

**Comment by RainerAtSpirit** [comment]
Here's another test that seems odd to me.
`area` is a computed, so why does it show up in the snapshot?

```javascript
    const ComputedFactory = types.model({
        width: 100,
        height: 200,
        get area(){
            return this.width * this.height
        }
    })
...
test("it should compose factories with simple computeds return types", (t) => {
    const {BoxFactory, ComputedFactory} = createTestFactories()
    const ComposedFactory = types.extend(BoxFactory, ComputedFactory)

    t.deepEqual(getSnapshot(ComposedFactory.create()), {width: 100, height: 200, area: 20000})
})
```

**Comment by mattiamanzati** [comment]
Uhm, I think that's due to https://github.com/mobxjs/mobx-state-tree/blob/master/src/types/complex-types/object.ts#L215, we need to actually filter properties with get descriptors out.

**Comment by RainerAtSpirit** [comment]
@mattiamanzati Instead of filtering them out would it be possible to attach all getters to the resultant factory?

```javascript
// Pseudo code ahead
const ComputedFactory = types.model({
    width: 50,
    height: 100,
    get area(){
        return this.width * this.height
    }
})

const ComplexComputedFactory = types.model({
    width: 100,
    height: 200,
    stringifiedJSON: '{"a":{"b":{"c":1,"d":2}}}',
    get jsonObj(){
        return JSON.parse(this.stringifiedJSON)
    }
})

const ComposedFactory = types.extend(ComputedFactory , ComplexComputedFactory )

// ComposedFactory should be equivalent to a factory that has been defined as 
const Factory= types.model({
    width: 100,
    height: 200,
    stringifiedJSON: '{"a":{"b":{"c":1,"d":2}}}',
    get area(){
        return this.width * this.height
    },
    get jsonObj(){
        return JSON.parse(this.stringifiedJSON)
    }
})

```

**Comment by mattiamanzati** [comment]
Should be fixed in 0.6.3, feel free to reopen if not working :)
